### ISSUE TITLE
Not able to replicate the HRFuser results on nuScenes

### ISSUE THREAD

**Issue by kvnptl** [timbroed/HRFuser#3]
Hello,

First of all, thank you for open-sourcing the code and providing an easy-to-follow README for testing the HRFuser model.

### Issue Summary:

I have been attempting to replicate the HRFuser results as per the HRFuser paper but have faced discrepancies in the metrics on the nuScenes dataset.

### Environment:
- Multi-GPU: 4x NVIDIA V100
- Python 3.8
- pytorch 1.10.0
- torchvision 0.11.0
- cuda 10.2
- gcc 10.1.0
- Other hyperparameters and model settings (as per the code repository)

### Steps to Reproduce:

1. Cloned the repository.
2. Installed all dependencies.
3. Ran the training script with 12 epochs.
4. Repeated training with 60 epochs.

### Overall Results:

- **Model: [Tiny]**
 
First, I tried 12 epochs with batch size 12, as mentioned in the paper, with the same settings as given in the code with multi-GPU. 

  | AP   | AP (0.5) | AP (0.75) | AP (small) | AP (medium) | AP (large) |
  |------|----------|-----------|------------|-------------|------------|
  | 0.12 | 0.26     | 0.07      | 0.00       | 0.06        | 0.17       |

Then, I tried 60 epochs with batch size 12 but got around the same results. Even though loss keeps on decreasing, the overall object detection score is not improving! 

  | AP  | AP (0.5) | AP (0.75) | AP (small) | AP (medium) | AP (large) |
  |-----|----------|-----------|------------|-------------|------------|
  | 0.17| 0.33     | 0.14      | 0.00       | 0.07        | 0.21       |

- Loss curves (for 60 epochs),

![image](https://github.com/timbroed/HRFuser/assets/47410011/32a8ac7a-99e9-4422-898c-52e7137f1db1)
![image](https://github.com/timbroed/HRFuser/assets/47410011/edf80d34-88d6-4769-be80-2debc6fbcaa8)

-  **Model: [Base]**

Even for the Base model, for 60 epochs with batch size 12, I am getting similar low results as the Tiny model. 

| AP  | AP (0.5) | AP (0.75) | AP (small) | AP (medium) | AP (large) |
|-----|----------|-----------|------------|-------------|------------|
|0.16 | 0.30     | 0.15      | 0.00       | 0.08        | 0.20       |

- Loss curves

![image](https://github.com/timbroed/HRFuser/assets/47410011/0166905e-59f6-4175-82b8-5070af26db4f)
![image](https://github.com/timbroed/HRFuser/assets/47410011/a2f65977-5fb6-437d-9e73-467d52924086)

###   With given pre-trained weights,

However, with the given weights, it's giving good results compared to the above models training from scratch. 

| AP  | AP (0.5) | AP (0.75) | AP (small) | AP (medium) | AP (large) |
|-----|----------|-----------|------------|-------------|------------|
| 0.41| 0.69     | 0.42      | 0.02       | 0.22        | 0.51       |

### Questions:

- Are there additional parameters or configurations that I may have missed?
- How can the loss decrease without corresponding improvements in the overall object detection scores?
- Are there any special tips for training these models from scratch to get good results?

Please let me know if any other info would be helpful for debugging.

Any help in debugging the issue would be greatly appreciated.

Thank you!

**Comment by tbroed** [comment]
As you closed the issue, did you fix the problem?

**Comment by kvnptl** [comment]
Thank you for the reply. 

It was my mistake. I initially started HRFuser with the nuScenes-mini dataset for basic tests.  However, I later forgot to download the full dataset and kept on doing all the training work on the mini dataset. So that's why, in the graphs, it is visible that the number of iterations is much less even with a batch size of 12.

So, now I am working with the full dataset. I'll update here if something goes wrong.
